### PR TITLE
Update testing docs.

### PR DESCRIPTION
**Pull request by jsha**
Recommend faster-running commands with clearer output first, building up to
slow, noisy, `tox` last.

Remove outdated Boulder testing advice, and outdated Boulder testing scripts
(Vagrantfile, mac-bootstrap.sh, boulder-start.sh). Update Boulder testing
section.

Fill out "directory structure" section of contributing guide a little better.

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -54,103 +54,58 @@ where appropriate.
...
-  ./tests/boulder-integration.sh
-
-Otherwise, please follow the following instructions.
+To run integration tests locally, you need Docker and docker-compose installed
+and working. Fetch and start Boulder using:
 
-macOS users: Run ``./tests/mac-bootstrap.sh`` instead of
-``boulder-start.sh`` to install dependencies, configure the
-environment, and start boulder.
+  ./tests/boulder-fetch.sh
```
Comment (ohemorange) [inline docs/contributing.rst]: this script doesn't actually work - it's missing a `BOULDERPATH=boulder` line, and when it fails it leaves the `boulder` directory there so you can't just run it again without deleting that directory first.

File: docs/contributing.rst [hunk 2]
```diff
@@ -54,103 +54,58 @@ where appropriate.
...
-  ./tests/boulder-integration.sh && echo OK || echo FAIL
-
-If you would like to test `certbot_nginx` plugin (highly
-encouraged) make sure to install prerequisites as listed in
-``certbot-nginx/tests/boulder-integration.sh`` and rerun
-the integration tests suite.
-
-.. _Boulder: https://github.com/letsencrypt/boulder
-.. _Go: https://golang.org
+  ./tests/boulder-integration.sh
```
Comment (ohemorange) [inline docs/contributing.rst]: This also doesn't seem to be working? I'm getting `ConnectionError: HTTPConnectionPool(host='localhost', port=4000): Max retries exceeded with url: /directory (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f3f07af4210>: Failed to establish a new connection: [Errno 111] Connection refused',))`

File: docs/contributing.rst [hunk 3]
```diff
@@ -54,103 +54,58 @@ where appropriate.
...
-- ``tox -e lint`` checks the style of the whole project, while
-  ``pylint --rcfile=.pylintrc path`` will check a single file or
-  specific directory only.
-
-- For debugging, we recommend ``pip install ipdb`` and putting
-  ``import ipdb; ipdb.set_trace()`` statement inside the source
-  code. Alternatively, you can use Python's standard library `pdb`,
-  but you won't get TAB completion...
-
+When you are working in a file foo.py, there should also be a file foo_test.py
```
Comment (ohemorange) [inline docs/contributing.rst]: "either in the same directory as ``foo.py`` or in the ``tests`` subdirectory."

also, `foo.py` and `foo_test.py` should have double quotes.

File: docs/contributing.rst [hunk 4]
```diff
@@ -54,103 +54,58 @@ where appropriate.
...
+(if there isn't, make one). While you are working on your code and tests, run
+``python foo_test.py`` to run the relevant tests.
+
+For debugging, we recommend ``pip install ipdb`` and putting
```
Comment (ohemorange) [inline docs/contributing.rst]: "we recommend running"

File: docs/contributing.rst [hunk 5]
```diff
@@ -54,103 +54,58 @@ where appropriate.
...
+``import ipdb; ipdb.set_trace()`` statement inside the source
```
Comment (ohemorange) [inline docs/contributing.rst]: statements

File: docs/contributing.rst [hunk 6]
```diff
@@ -54,103 +54,58 @@ where appropriate.
...
+code. Alternatively, you can use Python's standard library `pdb`,
+but you won't get TAB completion.
+
+Once you are done with your code changes, and the tests in foo_test.py pass,
```
Comment (ohemorange) [inline docs/contributing.rst]: ``foo_test.py``

File: docs/contributing.rst (hunk 1 above)
Comment (jsha) [inline docs/contributing.rst]: Sorry, I should have said in the PR description: These test instructions assume https://github.com/certbot/certbot/pull/4180 is landed.

File: docs/contributing.rst [hunk 7]
```diff
@@ -54,112 +54,76 @@ where appropriate.
...
-If you would like to test `certbot_nginx` plugin (highly
-encouraged) make sure to install prerequisites as listed in
-``certbot-nginx/tests/boulder-integration.sh`` and rerun
-the integration tests suite.
-
-.. _Boulder: https://github.com/letsencrypt/boulder
-.. _Go: https://golang.org
+  ./tests/boulder-integration.sh
 
+:: _removing all containers and volumes: https://www.digitalocean.com/community/tutorials/how-to-remove-docker-images-containers-and-volumes
```
Comment (ohemorange) [inline docs/contributing.rst]: This should be `..` not `::`

File: docs/contributing.rst [hunk 8]
```diff
@@ -54,112 +54,76 @@ where appropriate.
...
+and working. Fetch and start Boulder using:
 
-These commands download `Go`_ 1.5.3 to ``/tmp/``, extracts to ``/usr/local``,
-and then adds the export lines required to execute ``boulder-start.sh`` to
-``~/.profile`` if they were not previously added
+  ./tests/boulder-fetch.sh
 
-Make sure you execute the following command after `Go`_ finishes installing::
+If you have problems with Docker, you may want to try `removing all containers and
+volumes`_.
```
Comment (ohemorange) [inline docs/contributing.rst]: maybe "and making sure you have enough memory on the machine"?

**Comment by ohemorange** [comment]
Travis?

**Pull request merged**